From 099143e662df5be7689fb3c1a884ef7b8a27e8b2 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 15 Jun 2023 13:34:46 +0800
Subject: [PATCH] pfeng-slave: fix Unknown sysmbol do_exit error

In mainline commit eb55e716ac1aa ("exit: Stop exporting do_exit"),
do_exit is not exported anymore and this will cause that the modules
which use do_exit can't be inserted anymore with the error log:

pfeng_slave: Unknown symbol do_exit (err -2)
modprobe: ERROR: could not insert 'pfeng_slave': Unknown symbol in module, or unknown parameter (see dmesg)

Use kthread_complete_and_exit to take place of do_exit to fix it.

Upstream-Status: Pending

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 sw/linux-pfeng/pfeng-slave-drv.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sw/linux-pfeng/pfeng-slave-drv.c b/sw/linux-pfeng/pfeng-slave-drv.c
index 4a497ce..8a4e5c3 100644
--- a/sw/linux-pfeng/pfeng-slave-drv.c
+++ b/sw/linux-pfeng/pfeng-slave-drv.c
@@ -291,7 +291,7 @@ static int pfeng_drv_deferred_probe(void *arg)
 		while(1) {
 
 			if(kthread_should_stop())
-				do_exit(0);
+				kthread_complete_and_exit(NULL, 0);
 
 			if (hal_ip_ready_get())
 				break;
@@ -300,7 +300,7 @@ static int pfeng_drv_deferred_probe(void *arg)
 				/* Timed out */
 				HM_MSG_DEV_ERR(dev, "PFE controller UP timed out\n");
 				priv->deferred_probe_task = NULL;
-				do_exit(0);
+				kthread_complete_and_exit(NULL, 0);
 			}
 
 			usleep_range(100, 500);
@@ -403,7 +403,7 @@ static int pfeng_drv_deferred_probe(void *arg)
 
 	if (priv->deferred_probe_task) {
 		priv->deferred_probe_task = NULL;
-		do_exit(0);
+		kthread_complete_and_exit(NULL, 0);
 	}
 
 	return 0;
@@ -411,7 +411,7 @@ static int pfeng_drv_deferred_probe(void *arg)
 err_drv:
 	if (priv->deferred_probe_task) {
 		priv->deferred_probe_task = NULL;
-		do_exit(0);
+		kthread_complete_and_exit(NULL, 0);
 	}
 
 	return ret;
-- 
2.40.0

